mem_18.go

```diff
@@ -0,0 +1,10 @@
+// +build go1.8
+
+package capnp
+
+import "net"
+
+func (e *Encoder) write(bufs [][]byte) error {
+	_, err := (*net.Buffers)(&bufs).WriteTo(e.w)
```

Comment:
It's a result of how `net.Buffers` is implemented so it's very intentional. `net.Buffers` needs to work with multiple calls to `Read` and `WriteTo` (for things like `io.Copy`), to do this it needs to be able to modify the slice as it consumes more data - see [src/net/net.go](https://github.com/golang/go/blob/25a1d5d0f472f0da82aa8b4a1f3322463df5bbe6/src/net/net.go#L677).